src/Infrastructure/Elastic/ElasticIndexAdapter.php

```diff
@@ -22,9 +24,43 @@ public function create(IndexConfigurationInterface $indexConfiguration): void
         $this->client->indices()->create($indexConfiguration->toArray());
     }
 
+    public function pointToAlias(IndexConfigurationInterface $indexConfiguration): void
+    {
+        if (!$indexConfiguration->isAliased()) {
+            return;
+        }
+
+        $this->upsertAlias($indexConfiguration->getAliasConfiguration());
+
+        if($indexConfiguration->getAliasConfiguration()->shouldOldAliasesBePruned()) {
+            $this->pruneAliases($indexConfiguration->getAliasConfiguration());
+        }
+    }
+
     public function delete(IndexConfigurationInterface $indexConfiguration): void
     {
-        $this->client->indices()->delete(['index' => $indexConfiguration->getName()]);
+        $aliasConfiguration = $indexConfiguration->getAliasConfiguration();
+
+        if(!$indexConfiguration->isAliased()) {
+            $this->client->indices()->delete(['index' => $indexConfiguration->getName()]);
+            return;
+        }
+
+        $exists = $this->client->indices()->existsAlias(['name' => $aliasConfiguration->getAliasName()]);
+
+        if (!$exists) {
+            $this->client->indices()->putAlias([
```

Comment:
Why create an alias if your goal is to delete the index configuration and delete it at line 60?

Reply:
I don't get the question.

Reply:
In this function (`delete`)  you want to delete an index. But this line creates an alias if it doesn't exist. And on line 60 it get's deletes again.

Why create it in the first place?

Reply:
I think it was a misplaced copy-paste. Removed it.